### PR TITLE
refactor(honey): cleared some code redundancy

### PR DESCRIPTION
**Pull request by cypherpepe**
**apps/honey/src/utils/graph-utils.ts**
Instead of having separate functions for generating hourly and daily timestamps (fullHourTimestamps and fullDayTimestamps), we combined them into a single function generateTimestamps, which accepts a time step (e.g., 3600 for hourly and 86400 for daily).

The logic for populating volume and supply data (by hour and day) has been combined into a single, reusable function fillData. This function takes arguments like the data list, time step, type name, and a function to generate IDs. This eliminates the redundancy of separate functions for handling different cases.

The logic for processing hourly and daily data for both volume and supply is now handled by the same fillData function, removing duplicate code that was previously present for similar tasks.

By using one function for both volume and supply data handling, the code becomes more readable, easier to maintain, and scalable if future time intervals or data types need to be added.

**apps/honey/src/hooks/useHoneyEvents.ts**
A new helper function fetchHoneyData was created to handle the common logic for fetching transaction data.
The logic inside the useSWRInfinite hooks for fetching allData, mintData, and redemptionData was refactored to use this fetchHoneyData function, reducing code duplication.

**apps/honey/src/app/api/useHoneyEvents.ts**
1. Created a separate fetchData function to handle the repeated logic of making a fetch request to the API. This function accepts a URL, sends a GET request with headers, and returns the parsed JSON response or null in case of an error.
This reduces redundancy in the code since the fetch logic was used multiple times in each data handler (allData, mintData, burnData).

2. Adjusted the page number calculation by ensuring that the current page is correctly incremented with key[1] ?? 0 + 1 instead of key[1] + 1. This ensures that the pagination works correctly when fetching data.

### CODE REVIEW

File: apps/honey/src/hooks/useHoneyEvents.ts [hunk 1]
```diff
@@ -8,6 +8,26 @@ import useSWRInfinite from "swr/infinite";
 
 const DEFAULT_SIZE = 10;
 
+const fetchHoneyData = (query: any, txnType?: string) => {
```
Comment (bearpong) [inline apps/honey/src/hooks/useHoneyEvents.ts:11]: can you rename this to `generateHoneyDataFetcher`?  I think it's easier to understand

File: apps/honey/src/hooks/useHoneyEvents.ts [hunk 2]
```diff
@@ -8,6 +8,26 @@ import useSWRInfinite from "swr/infinite";
...
+  return async (key: any[]) => {
+    const page = key[1];
+    try {
+      const variables = txnType
+        ? { page: page * DEFAULT_SIZE, limit: DEFAULT_SIZE, txnType }
+        : { page: page * DEFAULT_SIZE, limit: DEFAULT_SIZE };
```
Comment (bearpong) [inline apps/honey/src/hooks/useHoneyEvents.ts:17]: why not just `const variables = { page: page * DEFAULT_SIZE, limit: DEFAULT_SIZE, txnType }`?

**Review by bearpong — CHANGES_REQUESTED** [review]
Hey @cypherpepe, good job on this PR. 

Can you please make some small changes?

File: apps/honey/src/hooks/useHoneyEvents.ts (hunk 1 above)
Comment (cypherpepe) [reply apps/honey/src/hooks/useHoneyEvents.ts]: I've renamed the function from fetchHoneyData to generateHoneyDataFetcher, as per your suggestion. It should be more intuitive now.

File: apps/honey/src/hooks/useHoneyEvents.ts (hunk 2 above)
Comment (cypherpepe) [reply apps/honey/src/hooks/useHoneyEvents.ts]: I’ve updated the variables declaration to a single object, removing the conditional logic, as suggested.

File: apps/honey/src/utils/graph-utils.ts [hunk 3]
```diff
@@ -1,111 +1,65 @@
...
-  }
-  return timestamps;
-}
-
-export function fillVolumeDataByHour(list: HoneyEntry[], endUnix: number) {
+function fillData(
+  list: HoneyEntry[],
+  endUnix: number,
+  step: number,
+  typename: string,
```
[the remark this answers is not in the document]
Comment (cypherpepe) [reply apps/honey/src/utils/graph-utils.ts]: The typename parameter has been typed explicitly, and now it accepts specific types

File: apps/honey/src/utils/graph-utils.ts [hunk 4]
```diff
@@ -1,111 +1,65 @@
...
+  getId: (time: number) => string,
```
[the remark this answers is not in the document]
Comment (cypherpepe) [reply apps/honey/src/utils/graph-utils.ts]: You were right, the getId function was unnecessary. I’ve removed it and now generate the ID directly using (time / step).toString() inside the fillData function.

**Review by bearpong — APPROVED** [review]

**Pull request merged**